qiskit_experiments/framework/base_experiment.py

```diff
@@ -512,46 +511,17 @@ def set_analysis_options(self, **fields):
     def _metadata(self) -> Dict[str, any]:
         """Return experiment metadata for ExperimentData.
 
-        The :meth:`_add_job_metadata` method will be called for each
-        experiment execution to append job metadata, including current
-        option values, to the ``job_metadata`` list.
+        Subclasses can override this method to add custom experiment
+        metadata to the returned experiment result data.
         """
-        metadata = {
-            "experiment_type": self._type,
-            "num_qubits": self.num_qubits,
-            "physical_qubits": list(self.physical_qubits),
-        }
-        # Add additional metadata if subclasses specify it
-        for key, val in self._additional_metadata().items():
-            metadata[key] = val
-        return metadata
-
-    def _additional_metadata(self) -> Dict[str, any]:
-        """Add additional subclass experiment metadata.
-
-        Subclasses can override this method if it is necessary to store
-        additional experiment metadata in ExperimentData.
-        """
-        return {}
+        metadata = {"physical_qubits": list(self.physical_qubits)}
 
-    def _add_job_metadata(self, metadata: Dict[str, Any], jobs: BaseJob, **run_options):
-        """Add runtime job metadata to ExperimentData.
-
-        Args:
-            metadata: the metadata dict to update with job data.
-            jobs: the job objects.
-            run_options: backend run options for the job.
-        """
-        values = {
-            "job_ids": [job.job_id() for job in jobs],
-            "experiment_options": copy.copy(self.experiment_options.__dict__),
-            "transpile_options": copy.copy(self.transpile_options.__dict__),
-            "run_options": copy.copy(run_options),
-        }
-        if self.analysis is not None:
-            values["analysis_options"] = copy.copy(self.analysis.options.__dict__)
-
-        metadata["job_metadata"] = [values]
+        # Store measurement level and meas return if they have been
+        # set for the experiment
+        for run_opt in ["meas_level", "meas_return"]:
+            if hasattr(self.run_options, run_opt):
+                metadata[run_opt] = getattr(self.run_options, run_opt)
+        return metadata
```

Comment:
My concern with this is that it is a half-solution. The data processing may require other run-options (ESP, delay's, qubit reset, etc...). In PR #689 you can see how such options are being used. In that PR we do not need them in the `get_processor` as above so the changes in #719 are compatible with #689. However, someone might later search for such run-options in the metadata. I suggest to either
* copy all run options to the metadata, or
* have a framework that does not need run options in the metadata at all.

In any case there should be a place to easily figure out what the run options were in the following situations:
* after the experiment has been run, and
* during the analysis step (since different run options will require a different data processing pipeline), and
* after loading a previously saved experiment.

Reply:
This metadata is only supposed to store run options needed for analysis. The developer of an experiment should overload their experiments `_metadata` method to add whatever values they want for analysis (and ideally use as little as possible). Currently these two run options are required by `get_processor` in majority of experiments (I could have not put this in base and instead overloaded all the curve analysis experiments to have this). If you want to change `get_processor/CurveAnalysis` so these are no longer needed they can be removed from here. If you need more later for restless processor you could add them explicitly to those experiments that require them.

The thing we want to avoid is storing all run options all the time in metadata because it results in too big files for DB and are not needed. For record keeping the goal for storing full run options should be be via an artifact of the full ExperimentConfig so you can recover the full experiment later, but this isn't intended for analysis. Any specific run options needed for analysis should be in metadata or set via analysis options.